ekncontent/ekncontent/eknc-domain.c

```diff
@@ -830,8 +858,13 @@ on_xapian_query_response (GObject *source,
           return;
         }
 
+      /* Note that we pass another closure here instead of just the task
```

Comment:
Okay, I've updated this so as to separate out the part which handles the "countdown" from the part that actually returns the task. All the countdown knows about is an opaque gpointer and a callback that gets called when all the child tasks are completed. The caller passes in a callback and the GTask and returns it accordingly.